vllm/entrypoints/chat_utils.py

```diff
@@ -401,29 +477,22 @@ def _parse_chat_message_content_parts(
 
     has_image = False
     for part in parts:
-        part_type = part["type"]
-        if part_type == "text":
-            text = _TextParser(part)["text"]
+        if isinstance(part, str):  # Handle plain text parts
+            text = _TextParser(part)
             texts.append(text)
-        elif part_type == "image_url":
-            image_url = _ImageParser(part)["image_url"]
-
-            if image_url.get("detail", "auto") != "auto":
-                logger.warning(
-                    "'image_url.detail' is currently not supported and "
-                    "will be ignored.")
-
-            mm_parser.parse_image(image_url["url"])
-            has_image = True
-        elif part_type == "audio_url":
-            audio_url = _AudioParser(part)["audio_url"]
-
-            mm_parser.parse_audio(audio_url["url"])
-        elif part_type == "refusal":
-            text = _RefusalParser(part)["refusal"]
-            texts.append(text)
-        else:
-            raise NotImplementedError(f"Unknown part type: {part_type}")
+        else:  # Handle structured dictionary parts
+            assert isinstance(part, dict)
+            part_type, content = _parse_chat_message_content_mm_part(part)
+
+            if part_type in ["text", "refusal"]:
```

Comment:
Use tuple instead of list is a better practice in terms of performance for a static list.

```suggestion
            if part_type in ("text", "refusal"):
```